src/mutate/drawDefinitions/positionGovernor/qualifierProgression.ts

```diff
@@ -0,0 +1,163 @@
+      structureId: sourceLink.source.structureId,
+      drawDefinition,
+    });
+
+    if (!requireCompletedStructures || structureCompleted) {
+      const qualifyingRoundNumber = structure.qualifyingRoundNumber;
+      const { matchUps } = getAllStructureMatchUps({
+        matchUpFilters: {
+          ...(qualifyingRoundNumber && {
+            roundNumbers: [qualifyingRoundNumber],
+          }),
+          hasWinningSide: true,
+        },
+        afterRecoveryTimes: false,
+        inContext: true,
+        structure,
+      });
+
+      for (const matchUp of matchUps) {
+        const winningSide = matchUp.sides.find((side) => side?.sideNumber === matchUp.winningSide);
+        const relevantSide = matchUp.matchUpStatus === BYE && matchUp.sides?.find(({ participantId }) => participantId);
+
+        if (winningSide || relevantSide) {
+          const { participantId } = winningSide || relevantSide || {};
+          if (participantId && !assignedParticipantIds.includes(participantId)) {
+            qualifyingParticipantIds.push(participantId);
+          }
+        }
+      }
+    }
+  }
+
+  for (const sourceLink of roundRobinSourceLinks) {
+    const structure = drawDefinition?.structures?.find(
+      (structure) => structure.structureId === sourceLink.source.structureId,
+    );
+    if (structure?.stage !== QUALIFYING) continue;
+
+    const structureCompleted = isCompletedStructure({
+      structureId: sourceLink.source.structureId,
+      drawDefinition,
+    });
+
+    if (!requireCompletedStructures || structureCompleted) {
+      const { positionAssignments } = getPositionAssignments({ structure });
+      const relevantParticipantIds: any =
+        positionAssignments
+          ?.map((assignment) => {
+            const participantId = assignment.participantId;
+            const results = findExtension({
+              element: assignment,
+              name: TALLY,
+            }).extension?.value;
+
+            return results ? { participantId, groupOrder: results?.groupOrder } : {};
+          })
+          .filter(
+            ({ groupOrder, participantId }) => groupOrder === 1 && !assignedParticipantIds.includes(participantId),
+          )
+          .map(({ participantId }) => participantId) ?? [];
+
+      if (relevantParticipantIds) qualifyingParticipantIds.push(...relevantParticipantIds);
+    }
+  }
+
+  if (!qualifyingParticipantIds.length) return decorateResult({ result: { error: MISSING_QUALIFIED_PARTICIPANTS } });
+
+  qualifierPositions.forEach((position) => {
+    const randomParticipantId = randomPop(qualifyingParticipantIds);
+    randomParticipantId &&
+      qualifierDrawPositionAssignment({
+        qualifyingParticipantId: randomParticipantId,
+        tournamentRecord,
+        drawDefinition,
+        drawPosition: position.drawPosition,
+        structureId: structure.structureId,
+      });
+  });
+
+  return decorateResult({
+    result: definedAttributes({
```

Comment:
I would like to return an array of `assignedParticipantIds` 
It doesn't look like `qualifiedDrawPositionAssignment` returns the position to which the assignment was made...